projects/packages/search/src/inline-search/class-inline-search-highlighter.php

```diff
@@ -0,0 +1,345 @@
+			return $content;
+		}
+
+		if ( ! empty( $this->highlighted_content[ $post_id ]['content'] ) ) {
+			// Apply wpautop to maintain paragraph formatting.
+			return wpautop( $this->highlighted_content[ $post_id ]['content'] );
+		}
+
+		return $content;
+	}
+
+	/**
+	 * Filter comment text to show highlighted version.
+	 *
+	 * @param string $comment_text The comment text.
+	 *
+	 * @return string The filtered comment text.
+	 */
+	public function filter_highlighted_comment( string $comment_text ): string {
+		if ( ! is_search() || ! in_the_loop() ) {
+			return $comment_text;
+		}
+
+		$post_id = get_the_ID();
+
+		if ( ! $this->is_search_result( $post_id ) || empty( $this->highlighted_content[ $post_id ]['comments'] ) ) {
+			return $comment_text;
+		}
+
+		return $this->highlighted_content[ $post_id ]['comments'];
+	}
+
+	/**
+	 * Process highlighting data for a single search result.
+	 *
+	 * @param array $result  The search result data from the API.
+	 * @param int   $post_id The post ID for this result.
+	 */
+	private function process_result_highlighting( array $result, int $post_id ) {
+		if ( empty( $result['highlight'] ) ) {
+			return;
+		}
+
+		$title    = $this->extract_highlight_field( $result, 'title' );
+		$content  = $this->extract_highlight_field( $result, 'content' );
+		$comments = $this->extract_highlight_field( $result, 'comments' );
+
+		$this->highlighted_content[ $post_id ] = array(
+			'title'    => $title,
+			'content'  => $content,
+			'comments' => $comments,
+		);
+	}
+
+	/**
+	 * Extract a highlight field from the search result, handling different field formats.
+	 *
+	 * @param array  $result The search result data from the API.
+	 * @param string $field  The field name to extract.
+	 *
+	 * @return string The extracted highlighted field.
+	 */
+	private function extract_highlight_field( array $result, string $field ): string {
+		// Try exact match first
+		if ( isset( $result['highlight'][ $field ] ) && is_array( $result['highlight'][ $field ] ) ) {
+			return $result['highlight'][ $field ][0];
+		}
+
+		// Try field variants with suffixes (e.g., 'title.default')
+		foreach ( $result['highlight'] as $key => $value ) {
+			if ( strpos( $key, $field . '.' ) === 0 ) {
+				if ( is_array( $value ) && ! empty( $value ) ) {
+					return $value[0];
+				}
+			}
+		}
+
+		return '';
+	}
+
+	/**
+	 * Check if the current post is a search result from our API
+	 *
+	 * @param int $post_id The post ID to check.
+	 *
+	 * @return bool Whether the post is a search result.
+	 */
+	public function is_search_result( int $post_id ): bool {
+		return is_search() && in_the_loop() && ! empty( $this->search_result_ids ) && in_array( $post_id, $this->search_result_ids, true );
+	}
+
+	/**
+	 * Get the highlighted content for a post.
+	 *
+	 * @param int $post_id The post ID.
+	 *
+	 * @return array|null The highlighted content array or null if not found.
+	 */
+	public function get_highlighted_content( int $post_id ): ?array {
```

Comment:
Maybe don't bother with this getter method since it isn't getting called from outside of this class.